myia/frontends/pytorch_functions.py

```diff
@@ -440,6 +469,74 @@ def nll_loss(logs, targets, reduction='mean'):
     return out
 
 
+# TODO
+# Parameter p does not currently support string values ('fro' and 'nuc')
+# (issues: p expected either number and not string, or string and not number)
+# Frobenius norm (p = 'fro') can be computed with p = 2.
+# Nuclear norm (p = 'nuc') is not yet supported.
+@core
+def norm(inp, p=None, dim=None):
+    """Map of torch.norm method."""
+    if p is None:
+        p = 2
+
+    # Convert p to input type.
+    p = torch.zeros(1, dtype=inp.dtype) + inp.dtype(p)
+    # Create a value 1 with input type.
+    one = torch.zeros(1, dtype=inp.dtype) + inp.dtype(1)
```

Comment:
Why do you add a zero?  `inp.dtype(x)` should already cast to the correct type.

Reply:
@abergeron It's because I got an error when using just scalars here at some points in pytorch backend (log below). So I just created these values as an array (with just 1 value) instead of as scalar to bypass the issue.

```
(myia) notoraptor@notoraptor-linux:~/mila/dev/git/myia$ pytest -xvvs tests/frontends/test_pytorch_ops.py::test_torch_norm
=========================================================================================== test session starts ============================================================================================
platform linux -- Python 3.7.5, pytest-5.3.2, py-1.8.0, pluggy-0.13.1 -- /home/notoraptor/anaconda3/envs/myia/bin/python
cachedir: .pytest_cache
rootdir: /media/win/Users/notoraptor/mila/dev/git/myia, inifile: pytest.ini
plugins: cov-2.8.1
collected 18 items                                                                                                                                                                                         

tests/frontends/test_pytorch_ops.py::test_torch_norm[relay-cpu-run0] <- tests/multitest.py PASSED
tests/frontends/test_pytorch_ops.py::test_torch_norm[pytorch-cpu-run0] <- tests/multitest.py PASSED
tests/frontends/test_pytorch_ops.py::test_torch_norm[relay-cpu-run1] <- tests/multitest.py PASSED
tests/frontends/test_pytorch_ops.py::test_torch_norm[pytorch-cpu-run1] <- tests/multitest.py PASSED
tests/frontends/test_pytorch_ops.py::test_torch_norm[relay-cpu-run2] <- tests/multitest.py PASSED
tests/frontends/test_pytorch_ops.py::test_torch_norm[pytorch-cpu-run2] <- tests/multitest.py FAILED

================================================================================================= FAILURES =================================================================================================
____________________________________________________________________________________ test_torch_norm[pytorch-cpu-run2] _____________________________________________________________________________________

test = <tests.multitest.MyiaFunctionTest object at 0x7fd52c908ad0>

    def runtest(test):
>       test.run(fn)

tests/multitest.py:66: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/multitest.py:166: in run
    return self.runtest(self, fn, **self.spec)
tests/multitest.py:280: in _run
    self.check(out, args, result)
tests/multitest.py:118: in check
    res = run(args)
tests/multitest.py:274: in out
    rval = mfn['output'](*args)
myia/pipeline/steps.py:474: in wrapped
    res = fn(*args)
myia/compile/channel/__init__.py:82: in __call__
    return self.channel.call_handle(self, args, kwargs)
myia/compile/channel/__init__.py:136: in call_handle
    return self._read_msg()
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = <myia.compile.channel.RPCProcess object at 0x7fd52ba04d90>

    def _read_msg(self):
        RemoteHandle.current_channel = self
        try:
            res = self.loader.get_data()
        finally:
            RemoteHandle.current_channel = None
        if isinstance(res, LoadedError):
>           raise res
E           myia.utils.serialize.LoadedError: Traceback (most recent call last):
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/channel/__main__.py", line 51, in _rpc_server
E               res = arg[0](*arg[1], **arg[2])
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/vm.py", line 90, in __call__
E               return wrap_result(self.eval(args))
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/vm.py", line 112, in eval
E               impl(*instr[1:])
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/vm.py", line 354, in inst_external
E               outs = fn(*(self._ref(a) for a in args))
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/backends/pytorch.py", line 408, in <lambda>
E               return lambda v: (backend.from_numpy(v),), [op.inputs[1]], [op]
E           
E             File "/media/win/Users/notoraptor/mila/dev/git/myia/myia/compile/backends/pytorch.py", line 445, in from_numpy
E               return torch.from_numpy(a).to(self.device)
E           
E           TypeError: expected np.ndarray (got numpy.float32)

myia/compile/channel/__init__.py:148: LoadedError
!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!! stopping after 1 failures !!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!
======================================================================================= 1 failed, 5 passed in 38.27s =======================================================================================
```

Reply:
Ok so that should use scalar_to_array after the cast then.  The array type you need is in pytorch_abstract_types.py (`APT`).

Reply:
@abergeron Done ! I also rebased the PR.